### PR TITLE
Create event when entry is pruned

### PR DESCRIPTION
**Pull request by faisal-memon**
**Pull Request check list**

- [ ] Commit conforms to CONTRIBUTING.md?
- [ ] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**
Events based cache

**Description of change**
Create an event when entries are pruned due to being expired.

**Which issue this PR fixes**
N/A

### CODE REVIEW

File: pkg/server/datastore/sqlstore/sqlstore.go [hunk 1]
```diff
@@ -3818,6 +3818,9 @@ func pruneRegistrationEntries(tx *gorm.DB, expiresBefore time.Time, logger logru
 		if err := deleteRegistrationEntrySupport(tx, entry); err != nil {
 			return err
 		}
+		if err := createRegistrationEntryEvent(tx, entry.EntryID); err != nil {
```
Comment (zmt) [inline pkg/server/datastore/sqlstore/sqlstore.go:3821]: Can we add coverage for actually pruning events? It looks like this test case prunes zero and expects zero entries: https://github.com/spiffe/spire/blob/main/pkg/server/datastore/sqlstore/sqlstore_test.go#L4028-L4032

Comment (faisal-memon) [reply pkg/server/datastore/sqlstore/sqlstore.go:3821]: updated existing unit test to cover this case. lmk if it looks ok to you.

**Review by azdagron — APPROVED** [review]
Thanks @faisal-memon

**Comment by amartinezfayo** [comment]
Hi @faisal-memon, sign-off is missing is missing in one commit:
```
Commit sha: [98db1cf](https://github.com/spiffe/spire/pull/4860/commits/98db1cfba69182ba1a928cdc94fa0c7f46d371df), Author: Faisal Memon, Committer: Faisal Memon; The sign-off is missing.
```
Could you please take a look at that?

**Comment by faisal-memon** [comment]
dco fixed

File: pkg/server/datastore/sqlstore/sqlstore.go (hunk 1 above)
Comment (zmt) [reply pkg/server/datastore/sqlstore/sqlstore.go:3821]: It's very similar to @stevend-uber's adjustment to this unit test. It covers event creation so, yes, looks OK. Thanks for updating.

**Review by zmt — APPROVED** [review]

File: pkg/server/datastore/sqlstore/sqlstore.go (hunk 1 above)
Comment (faisal-memon) [reply pkg/server/datastore/sqlstore/sqlstore.go:3821]: Thats good. Thanks for confirming
